sycl/source/CMakeLists.txt

```diff
@@ -147,6 +145,18 @@ function(add_sycl_rt_library LIB_NAME LIB_OBJ_NAME)
     PRIVATE OpenCL-Headers
   )
 
+  # Link with LLVMSupport and LLVMObject for shared utilities.
+  add_dependencies(${LIB_OBJ_NAME} LLVMSupport LLVMObject)
+  target_include_directories(${LIB_OBJ_NAME} PRIVATE ${LLVM_MAIN_INCLUDE_DIR})
```

Comment:
Is using `PRIVATE` keyword here enough to protect us from accidentally including LLVM headers into SYCL headers?

Reply:
Based on local testing, 384 check-sycl tests start failing with "fatal error: 'llvm/X.h' file not found" if I introduce an include of an LLVM header in the non-source headers.